log4j2-propagation/src/test/java/nl/talsmasoftware/context/log4j2/threadcontext/Log4J2ThreadContextSnapshotTest.java

```diff
@@ -193,92 +196,55 @@ void testApplyToCurrentThread() {
     }
 
-    /**
-     * Verify that {@link Log4j2ThreadContextData#applyToCurrentThread(Log4j2ThreadContextData, boolean)}
-     * with {@code overwrite=true} overwrites all existing data.
-     */
     @Test
-    void testApplyToCurrentThread_overwrite() {
-        ThreadContext.put("map1", "value1");
-        ThreadContext.put("map2", "value2");
-        ThreadContext.push("stack1");
-        ThreadContext.push("stack2");
+    void testImmutableContextMap() {
+        ThreadContext.put("key1", "value1");
+        ThreadContext.put("key2", "value2");
+        final Log4j2ThreadContextSnapshot snapshot = Log4j2ThreadContextSnapshot.captureFromCurrentThread();
 
-        Log4j2ThreadContextData data = Log4j2ThreadContextData.fromCurrentThreadContext();
-
-        ThreadContext.clearAll();
-        ThreadContext.put("map1", "old-value1");
-        ThreadContext.put("map3", "old-value3");
-        ThreadContext.push("stack3");
-
-        Log4j2ThreadContextData.applyToCurrentThread(data, true);
-
-        Map<String, String> expectedMap = new HashMap<String, String>();
-        expectedMap.put("map1", "value1");
-        expectedMap.put("map2", "value2");
-        assertEquals(expectedMap, ThreadContext.getContext());
-
-        List<String> expectedStack = Arrays.asList("stack1", "stack2");
-        assertEquals(expectedStack, ThreadContext.getImmutableStack().asList());
-    }
-
-    /**
-     * Verify that {@link Log4j2ThreadContextData#applyToCurrentThread(Log4j2ThreadContextData, boolean)}
-     * with {@code data=null,overwrite=false} has no effect.
-     */
-    @Test
-    void testApplyToCurrentThread_null() {
-        ThreadContext.put("map1", "value1");
-        ThreadContext.put("map2", "value2");
-        ThreadContext.push("stack1");
-        ThreadContext.push("stack2");
-
-        // Should have no effect
-        Log4j2ThreadContextData.applyToCurrentThread(null, false);
-
-        Map<String, String> expectedMap = new HashMap<String, String>();
-        expectedMap.put("map1", "value1");
-        expectedMap.put("map2", "value2");
-        assertEquals(expectedMap, ThreadContext.getContext());
-
-        List<String> expectedStack = Arrays.asList("stack1", "stack2");
-        assertEquals(expectedStack, ThreadContext.getImmutableStack().asList());
+        assertThrows(UnsupportedOperationException.class, new Executable() {
+            public void execute() {
+                snapshot.getContextMap().put("key3", "value3");
+            }
+        });
+        assertThat(snapshot.getContextMap(), aMapWithSize(2));
     }
 
-    /**
-     * Verify that {@link Log4j2ThreadContextData#applyToCurrentThread(Log4j2ThreadContextData, boolean)}
-     * with {@code data=null,overwrite=true} clears thread context.
-     */
     @Test
-    void testApplyToCurrentThread_null_overwrite() {
-        ThreadContext.put("map1", "value1");
-        ThreadContext.put("map2", "value2");
+    void testImmutableContextStack() {
```

Comment:
This test method seems to overlap with `test_umodifiableViews()` maybe it would be good to remove one of them?